### PR TITLE
Protecting remainingItemCount behind a feature flag. Also updating the API doc

### PR DESCRIPTION
**Pull request by caesarxuchao**
/kind bug
/sig api-machinery
/assign @smarterclayton @lavalamp 

Ref https://github.com/kubernetes/community/pull/3710
https://github.com/kubernetes/kubernetes/pull/75993#discussion_r283439706

I didn't get time to address this until now.

### CODE REVIEW

**Comment by smarterclayton** [comment]
This must be part of 1.15 because of the new feature introduced being behind a gated flag.

**Comment by timmycarr** [comment]
Hi! I'm a 1.15 Bug triage shadow and I'd like to check in on the status of this PR. Is this a release-blocking PR and is it still planned for 1.15?

**Comment by caesarxuchao** [comment]
@timmycarr thank you for checking. Yes, this is release blocking.

**Comment by soggiest** [comment]
Any updates on this? Who do we need to get this approved and merged if it's blocking the release?

File: test/e2e/apimachinery/chunking.go [hunk 1]
```diff
@@ -30,12 +30,18 @@ import (
...
 	"k8s.io/apiserver/pkg/storage/storagebackend"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	"k8s.io/client-go/util/workqueue"
 	"k8s.io/kubernetes/test/e2e/framework"
 	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
 )
 
+func shouldCheckRemainingItem() bool {
+	return utilfeature.DefaultFeatureGate.Enabled(features.RemainingItemCount)
+}
```
Comment (smarterclayton) [inline test/e2e/apimachinery/chunking.go]: This isn’t set magically by e2e reading from the server, right?

**Comment by smarterclayton** [comment]
One question

File: test/e2e/apimachinery/chunking.go (hunk 1 above)
Comment (caesarxuchao) [reply test/e2e/apimachinery/chunking.go]: No, it's local.

Comment (smarterclayton) [reply test/e2e/apimachinery/chunking.go]: Ok, and these aren’t conformance tests yet?

Comment (caesarxuchao) [reply test/e2e/apimachinery/chunking.go]: No. Conformance tests starts with "framework.ConformanceIt".

**Pull request merged**